### PR TITLE
Fix removing original SFTP server settings from memory

### LINKED ISSUE #1773: Crash when updating an SFTP Connection
**Describe the bug**
Amaze crashes with an IndexOutOfBoundException when trying to update an SFTP configuration.

**To Reproduce**
1. Create a new Cloud Connection
2. Select "SCP/SFTP Connection"
3. Insert a valid configuration
4. Save
5. Edit the same configuration and change IP, port and/or username
6. Update
7. Crash

**Expected behavior**
The configuration should be updated without any crash

**Smartphone (please complete the following information):**
 - Device: Moto Z Play
 - OS: Android 8
 - Rooted: No
 - Version: 3.3.4

**Additional context**
I've also reproduced it on the Android emulator.

The problem is caused by line 339 of `SftpConnectDialog`.
Changing any of the connection parameters (e.i. IP, port, username) also changes the value of `path`, which means the call to `containsServer` will return -1 and crash the following call to `removeServer`.
https://github.com/TeamAmaze/AmazeFileManager/blob/4da5bdcfd788a96865134e0837913ad8a2276622/app/src/main/java/com/amaze/filemanager/ui/dialogs/SftpConnectDialog.java#L338-L341

However, changing that line doesn't really solve the issue because the old configuration is left behind and it will only disappear after a restart.

I started working on a pull request (I hope it's welcome) aimed to fix this issue and a few other minor ones all related to updating a SFTP connection. I'll keep you updated.

### PR DESCRIPTION
**Pull request by TranceLove**
Fixes #1773.

- use original SSH URI when SftpConnectDialog is in edit mode
- validation routine that causes editing SSH connection settings unable to update due to credential (not)being updated
- pass argument to SftpConnectDialog using integer

### CODE REVIEW

**Comment by TheZxcv** [comment]
@TranceLove Hey, you beat me to it!
Anyway, there are still two related issues:
- Any update to the password isn't saved.
- The host fingerprint is never updated, this is an issue when changing the IP.

I have these fixes ready, I based them off this pull request.
I'm not quite sure what to do.

**Comment by TranceLove** [comment]
You can add your contribution to my pull request directly, or simply create a pull request directly instead, and I'll retract this once you created. Let's collaborate!

**Comment by TranceLove** [comment]
While checking updates from @TheZxcv I found my code still crash on LG Nexus 5x running AOSPExtended (9.0) for violating database unique constraints. Please wait before it's fixed...

**Comment by TranceLove** [comment]
As of b9aff11/7348be4 support for handling changed SSH server host key fingerprint had been added, lifting PR-WIP flag.

File: app/src/main/java/com/amaze/filemanager/ui/dialogs/SftpConnectDialog.java [hunk 1]
```diff
@@ -162,15 +163,44 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
             final String hostname = addressET.getText().toString();
             final int port = Integer.parseInt(portET.getText().toString());
             final String username = usernameET.getText().toString();
-            final String password = passwordET.getText() != null ?
-                    passwordET.getText().toString() : null;
+            final String password = !TextUtils.isEmpty(passwordET.getText()) ?
+                    passwordET.getText().toString() : getArguments().getString("password", null);
```
Comment (bowiechen) [inline app/src/main/java/com/amaze/filemanager/ui/dialogs/SftpConnectDialog.java]: I think we can simplify this logic into:
```Java
isEmpty(...) ? getArguments().getString("password", null) : passwordET.getText().toString();
```

to make things less complicated to maintain.

Comment (bowiechen) [inline app/src/main/java/com/amaze/filemanager/ui/dialogs/SftpConnectDialog.java]: I think we can simplify this logic into:
```Java
isEmpty(...) ? getArguments().getString("password", null) : passwordET.getText().toString();
```

to make things less complicated to maintain.

**Comment by EmmanuelMess** [comment]
@TranceLove please remember that translations will be overwritten.

**Comment by TranceLove** [comment]
> @TranceLove please remember that translations will be overwritten.

Acknowledged. So I'll fill in the blanks for zh-TW and ja-JP once PR cames up to transifex.

**Comment by EmmanuelMess** [comment]
> Acknowledged. So I'll fill in the blanks for zh-TW and ja-JP once PR cames up to transifex?

Yeah.

**Comment by EmmanuelMess** [comment]
Getting when updating name of ssh connection:

> 2019-12-04 15:25:17.671 7214-7214/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.amaze.filemanager.debug, PID: 7214
    android.os.NetworkOnMainThreadException
        at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1565)
        at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:116)
        at java.net.SocketOutputStream.write(SocketOutputStream.java:161)
        at net.schmizz.sshj.transport.TransportImpl.write(TransportImpl.java:451)
        at net.schmizz.sshj.transport.TransportImpl.sendDisconnect(TransportImpl.java:469)
        at net.schmizz.sshj.transport.TransportImpl.disconnect(TransportImpl.java:414)
        at net.schmizz.sshj.transport.TransportImpl.disconnect(TransportImpl.java:404)
        at net.schmizz.sshj.transport.TransportImpl.disconnect(TransportImpl.java:399)
        at net.schmizz.sshj.SSHClient.disconnect(SSHClient.java:440)
        at com.amaze.filemanager.filesystem.ssh.SshClientUtils.tryDisconnect(SshClientUtils.java:230)
        at com.amaze.filemanager.filesystem.ssh.SshConnectionPool.removeConnection(SshConnectionPool.java:96)
        at com.amaze.filemanager.ui.dialogs.SftpConnectDialog.lambda$onCreateDialog$8$SftpConnectDialog(SftpConnectDialog.java:176)
        at com.amaze.filemanager.ui.dialogs.-$$Lambda$SftpConnectDialog$_RbcaWUtNLsEbSmU5Q9RfFiNLU0.onClick(Unknown Source:16)
        at com.afollestad.materialdialogs.MaterialDialog.onClick(MaterialDialog.java:410)
        at android.view.View.performClick(View.java:7125)
        at android.view.View.performClickInternal(View.java:7102)
        at android.view.View.access$3500(View.java:801)
        at android.view.View$PerformClick.run(View.java:27336)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)

**Pull request merged**